src/utils/porn-socket/walltaker.tsx

```diff
@@ -0,0 +1,169 @@
+import { PropsWithChildren, useEffect, useRef } from 'react';
+import { PornSocketContext, PornSocketService } from './porn-socket-service.tsx';
+import { useImages } from '../../settings';
+import { ImageServiceType, ImageType } from '../../types';
+
+export interface LinkResponse {
+  id: number;
+  expires: string;
+  terms: string;
+  blacklist: string;
+  post_url: string;
+  post_thumbnail_url: string;
+  post_description: string;
+  created_at: string;
+  updated_at: string;
+  response_type?: any;
+  response_text?: any;
+  username: string;
+  online: boolean;
+}
+
+export interface UserResponse {
+  username: string;
+  id: number;
+  set_count: number;
+  online: boolean;
+  authenticated: boolean;
+  links: LinkResponse[];
+}
+
+export class WalltakerSocketService implements PornSocketService {
```

Comment:
lets move this into its own foilder, `walltalker`, together with its socket managment file and the `WalltalkerSettings`, just like the `e621` folder.

Reply:
Will do!